src/prpy/planning/vectorfield.py

```diff
@@ -197,32 +217,165 @@ def TerminateMove():
 
             return Status.CONTINUE
 
+        integration_timelimit = 10.0
         return self.FollowVectorField(robot, vf_straightline, TerminateMove,
+                                      integration_timelimit,
+                                      timelimit, 
+                                      **kw_args)
+
+
+    @PlanningMethod
+    def PlanWorkspacePath(self, robot, traj,
+                          timelimit=5.0,
+                          position_tolerance=0.01,
+                          angular_tolerance=0.15,
+                          goal_pose_tol=0.01,
+                          **kw_args):
+        """
+        Plan a configuration space path given a workspace path.
+        Trajectory timing information is ignored.
+
+        @param openravepy.Robot      robot: The robot.
+        @param openravepy.Trajectory traj:  Workspace trajectory,
+                                            represented as an
+                                            OpenRAVE AffineTrajectory.
+        @param float timelimit: Max planning time (seconds).
+        @param float position_tolerance: Constraint tolerance (meters).
+        @param float angular_tolerance:  Constraint tolerance (radians).
+        @param float goal_pose_tol: Goal tolerance (meters).
+
+        @return openravepy.Trajectory qtraj: Configuration space path.
+        """
+        if position_tolerance < 0.0:
+            raise ValueError('Position tolerance must be non-negative.')
+        elif angular_tolerance < 0.0:
+            raise ValueError('Angular tolerance must be non-negative.')
+
+        manip = robot.GetActiveManipulator()
+        Tstart = manip.GetEndEffectorTransform()
+
+        # Get the final end-effector pose
+        duration = traj.GetDuration()
+        T_ee_goal = openravepy.matrixFromPose(traj.Sample(duration)[0:7])
```

Comment:
This assumes that the trajectory has a single `AffineDOFs` group that is parameterized by a translation and a quaternion. If you are making this assumption, then you should check for this when you validate the input arguments (and add a few unit tests).